geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/messenger/JGroupsMessenger.java

```diff
@@ -335,7 +335,9 @@ public void start() throws MemberStartupException {
     try {
       Object oldDSMembershipInfo = services.getConfig().getOldMembershipInfo();
       if (oldDSMembershipInfo != null) {
-        logger.debug("Reusing JGroups channel from previous system", properties);
+        if (logger.isDebugEnabled()) {
```

Comment:
This extra conditional (and the others like it) don't seem to improve performance or maintainability. Is there a coding standard for Geode, that necessitates this pre-checking?

I searched the codebase for e.g. "logger.debug(" calls and while many do the pre-check like this, many others do not.

Reply:
I've changed my mind on this. I've learned that performance tests have shown that Geode's logging infrastructure should have conditional guards around log calls. A [Design Decision](https://cwiki.apache.org/confluence/display/GEODE/Design+Decisions) on this subject is coming from @kirklund that will describe the testing and the resulting conclusion.